app/models/viewmodels/NotificationPageSingleElementFactory.scala

```diff
@@ -28,20 +28,23 @@ import play.api.i18n.Messages
 import play.twirl.api.{Html, HtmlFormat}
 
 @Singleton
-class NotificationPageSingleElementFactory @Inject()(decoder: Decoder)(implicit messages: Messages) {
+class NotificationPageSingleElementFactory @Inject()(decoder: Decoder) {
 
   private val dateFormatter = DateTimeFormatter.ofPattern("dd MMM yyyy").withZone(ZoneId.systemDefault())
   private val timeFormatter = DateTimeFormatter.ofPattern("HH:mm").withZone(ZoneId.systemDefault())
 
-  def build(submission: SubmissionFrontendModel): NotificationsPageSingleElement = submission.actionType match {
-    case Arrival            => buildForRequest(submission)
-    case Departure          => buildForRequest(submission)
-    case DucrAssociation    => buildForDucrAssociation(submission)
-    case DucrDisassociation => buildForRequest(submission)
-    case ShutMucr           => buildForRequest(submission)
-  }
-
-  private def buildForRequest(submission: SubmissionFrontendModel): NotificationsPageSingleElement =
+  def build(submission: SubmissionFrontendModel)(implicit messages: Messages): NotificationsPageSingleElement =
+    submission.actionType match {
+      case Arrival            => buildForRequest(submission)
+      case Departure          => buildForRequest(submission)
+      case DucrAssociation    => buildForDucrAssociation(submission)
+      case DucrDisassociation => buildForRequest(submission)
+      case ShutMucr           => buildForRequest(submission)
```

Comment:
Probably worth to use `Arrival | Departure | DucrDisassociation | ShutMucr => buildForRequest(submission)`. What do you think?